src/main.cpp

```diff
@@ -3726,7 +3773,6 @@ bool ProcessNewBlock(CValidationState& state, const CChainParams& chainparams, c
       }
 
     LogPrint("thin", "Processing new block %s from peer %s (%d).\n", pblock->GetHash().ToString(), pfrom ? pfrom->addrName.c_str():"myself",pfrom ? pfrom->id: 0);
-    if (!IsInitialBlockDownload()) SendExpeditedBlock(*pblock,pfrom);
```

Comment:
this already gets called when we receive a normal block or xthin, why do
it twice?

On 04/09/2016 3:17 AM, gandrewstone wrote:
>
> In src/main.cpp
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/79#discussion_r77449973>:
>
> > @@ -3726,7 +3773,6 @@ bool ProcessNewBlock(CValidationState& state, const CChainParams& chainparams, c
> >        }
> >  
> >      LogPrint("thin", "Processing new block %s from peer %s (%d).\n", pblock->GetHash().ToString(), pfrom ? pfrom->addrName.c_str():"myself",pfrom ? pfrom->id: 0);
> > -    if (!IsInitialBlockDownload()) SendExpeditedBlock(*pblock,pfrom);
>
> why is the expedited block send deleted here?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/BitcoinUnlimited/BitcoinUnlimited/pull/79/files/efaaa2474cc017ebc1c84fa3748b50c293bff5c7#r77449973>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AMRF0BF51XDNqDG5uOxDbzm8hmdYe8lNks5qmpq8gaJpZM4JokbP>.
>